src/shared/services/guideResolverHelper.tsx

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React from 'react'
+import MdxSlide from 'browser/modules/Docs/MDX/MdxSlide'
+import Slide from 'browser/modules/Carousel/Slide'
+import docs, { isGuideChapter } from 'browser/documentation'
+import guideUnfound from 'browser/documentation/play-guides/unfound'
+import {
+  addProtocolsToUrlList,
+  extractAllowlistFromConfigString,
+  resolveAllowlistWildcard
+} from './utils'
+import { fetchRemoteGuide } from 'shared/modules/commands/helpers/play'
+import {
+  getDefaultRemoteContentHostnameAllowlist,
+  getRemoteContentHostnameAllowlist
+} from 'shared/modules/dbMeta/dbMetaDuck'
+import { splitMdxSlides } from 'browser/modules/Docs/MDX/splitMdx'
+import {
+  StyledCypherErrorMessage,
+  StyledDiv,
+  StyledErrorH4,
+  StyledHelpContent,
+  StyledHelpDescription,
+  StyledHelpFrame,
+  StyledPreformattedArea
+} from 'browser/modules/Stream/styled'
+
+const { chapters } = docs.guide
+const unfound = { slides: [guideUnfound.content], title: guideUnfound.title }
+
+export async function resolveGuide(
+  guideName: string,
+  store: any
```

Comment:
I think it'd be cleaner to just pass the state here instead of the store, since all of the downstream functions only seem to use `store.getState()`

Reply:
makes sense!